File: sesi17/bonusPegawai.js
console.log('program menghitung bonus pegawai')
const readline = require('node:readline').createInterface({
  input: process.stdin,
  output: process.stdout
})

readline.question('masukkan gaji pokok: ', gajiPokok => {
  readline.question('masukkan golongan pegawai: ', golPegawai => {
    readline.question('masukkan tahun lama bekerja: ', lamaKerja => {
      readline.question('masukkan umur pegawai: ', umur => {
        let bonus = 0
        // tulis logic disini
        if (golPegawai === 'staff') {
          if (lamaKerja > 5 && umur > 50) {
            bonus = 1000000
          } else {
            bonus = 50000
          }
        } else {
          if (lamaKerja > 5 && umur > 50) {
            bonus = 400000
          } else {
            bonus = 250000
          }
        }
        let totalPendapatan = Number(gajiPokok) + bonus
        console.log(`total pendapatan adalah ${totalPendapatan} rupiah`)
        readline.close()
      })
    })
  })
})

File: sesi17/bonus_pegawai.rb
puts "program menghitung bonus pegawai"
print "masukkan gaji pokok: "
gaji_pokok = gets.chomp.to_i
print "masukkan golongan pegawai: "
gol_pegawai = gets.chomp
print "masukkan tahun lama bekerja: "
lama_kerja = gets.chomp.to_i
print "masukkan umur pegawai: "
umur = gets.chomp.to_i
bonus = 0
# tulis logic disini
if gol_pegawai == "staff" # eksekusi pegawai staff
  if lama_kerja > 5 && umur > 50
    bonus = 1_000_000
  else
    bonus = 500_000
  end
else # eksekusi pegawai non-staff
  if lama_kerja > 5 && umur > 50
    bonus = 400_000
  else
    bonus = 250_000
  end
end
total_pendapatan = gaji_pokok + bonus
puts "total pendapatan adalah #{total_pendapatan} rupiah"
